File: iOS/RTCAudioLiveRoom/RTCAudioLiveRoom/RTC/Seats/SeatInfo.m
//
//  Seat.m
//  RTCAudioLiveRoom
//
//  Created by Aliyun on 2020/7/6.
//

#import "SeatInfo.h"

@interface SeatInfo ()

@end

@implementation SeatInfo

- (instancetype)init
{
    self = [super init];
    if (self) {
        self.seatIndex = -1;
        self.muteMic = NO;
        self.speaking = NO;
    }
    return self;
}
- (NSString *)icon
{
    if (_userId.length == 0)
    {
        _icon = @"empty";
    }else {
        _icon = [NSString stringWithFormat:@"%d",(int)_seatIndex + 1];
    }
    return _icon;
}

- (void)setUserId:(NSString *)userId
{
    _userId = userId;
    
    if (userId.length==0)
    {
        _icon = @"empty";
        _userName = [NSString stringWithFormat:@"%d号麦",(int)_seatIndex + 1];
        _speaking = NO;
        _muteMic = NO;
    }else {
        _icon = [NSString stringWithFormat:@"%d",(int)_seatIndex + 1];
    }
}
@end

File: android/RTCAudioLiveRoom/RTCAudioLiveRoom_Demo/src/main/java/com/aliyun/rtc/audiochatroom/bean/SeatInfo.java
package com.aliyun.rtc.audiochatroom.bean;

import android.support.annotation.Nullable;

import com.alivc.rtc.device.utils.StringUtils;

public class SeatInfo {

    private String seatIndex;
    private String userId;
    private String userName;
    private boolean muteMic;
    private boolean speaking;

    public String getSeatIndex() {
        return seatIndex;
    }

    public void setSeatIndex(String seatIndex) {
        this.seatIndex = seatIndex;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public boolean isMuteMic() {
        return muteMic;
    }

    public void setMuteMic(boolean muteMic) {
        this.muteMic = muteMic;
    }

    public boolean isSpeaking() {
        return speaking;
    }

    public void setSpeaking(boolean speaking) {
        this.speaking = speaking;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return obj instanceof SeatInfo && StringUtils.equals(((SeatInfo) obj).userId, this.userId);
    }
}
